lib/matplotlib/axes/_base.py

```diff
@@ -4173,6 +4231,7 @@ def twinx(self):
         self.yaxis.tick_left()
         ax2.xaxis.set_visible(False)
         ax2.patch.set_visible(False)
+        self._twinned_axes.join(self, ax2)
```

Comment:
Here and below, this `join` should not be needed; it is being done at the top of each function by the call to `_make_twin_axes`.